charmcraft/templates/init-machine/tests/unit/test_charm.py.j2

```diff
@@ -3,21 +3,22 @@
 #
 # Learn more about testing at: https://juju.is/docs/sdk/testing
 
-import unittest
-
 import ops
 import ops.testing
+import pytest
 from charm import {{ class_name }}
 
 
-class TestCharm(unittest.TestCase):
-    def setUp(self):
-        self.harness = ops.testing.Harness({{ class_name }})
-        self.addCleanup(self.harness.cleanup)
+@pytest.fixture
+def harness():
+    harness = ops.testing.Harness({{ class_name }})
+    yield harness
+    harness.cleanup()
```

Comment:
It would be nice to have this fixture be included upstream by making a pytest plugin as part of ops.

Reply:
You could indeed open a PR in pytest-operator and see if they are interested. Most charms currently use pytest-operator for integration tests so it wouldn't be an additional dependency for the project overall, although it probably would be for their unit tests.

I don't really see the value in putting a trivial 4-line fixture in a library personally. You're also likely going to need to provide custom metadata in some tests so will need alternative versions anyway.